src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -475,9 +475,15 @@ private void AppendLineNoPromptInjection(ITextBuffer buffer)
             }
 
             /// <summary>Implements <see cref="IInteractiveWindowOperations2.TypeChar"/>.</summary>
-            internal void TypeChar(char typedChar)
+            public void TypeChar(char typedChar)
             {
-                InsertText(typedChar.ToString());
+                using (var transaction = UndoHistory?.CreateTransaction(InteractiveWindowResources.TypeChar))
```

Comment:
Why are we implementing this at all? This will break merging of typed character undo operations, amongst other things. Looking at PR #6914, it's not clear to me why this approach is necessary. I would expect you'd still be deferring to the editor for the actual insertion handling, doing whatever setup you need before forwarding to the next handler.